src/routes/builder.tsx

```diff
@@ -20,16 +21,19 @@ export const Route = createFileRoute('/builder')({
       description: 'Build amazing applications with TanStack',
     }),
   }),
+  beforeLoad: async (ctx) => {
+    requireAuth(ctx)
```

Comment:
Utility that checks for a user id, and if none is present, throws a redirect with the current location in the `redirectTo` query param.